packages/ballerina-extension/src/stateMachine.ts

```
@@ -837,6 +826,52 @@ const stateMachine = createMachine<MachineContext>(
+            }
+            if (waitForReady) {
+                await visualizerWebviewReady;
+            }
+            resolve(true);
+        } catch (e) {
+            // Never silently: a failure here means no webview appears at all, which
+            // is exactly how the startup panel regressed unnoticed before.
+            console.error("Failed to open the visualizer webview panel.", e);
+            reject(e);
+        }
+    });
```

Biome (2.5.5)

[error] Promise executor functions should not be `async`.

(lint/suspicious/noAsyncPromiseExecutor)